Replace the underlying persistent store if incompatible with model (optional)

phchang opened the pull request.

Commit:
adding an option to overwrite the persistent store if model incompatibility is detected (vs. crashing or migrating)

Commit:
removing unneeded semicolons and replacing `++` with `+= 1` in preparation for Swift 3

Commit:
bumping pod to 1.0.0

tonystone commented on Pod/Cache/GenericCoreDataStack.swift:
```diff
@@ -185,14 +191,38 @@ public class GenericCoreDataStack<CoordinatorType: NSPersistentStoreCoordinator,
+                        let fileManager = NSFileManager.defaultManager()
+
+                        try fileManager.removeItemAtURL(storeURL)
+                        try fileManager.removeItemAtURL(NSURL(fileURLWithPath: "\(path)-shm"))
```
@phchang We should handle the case where the file may not be present, suppose there was a -wal but not -shm.  If the removeItemAtURL is tried on the -shm, it jump to the catch, skipping the -wal.

Would a better approach be to get a listing of files that match the filename <filename>.\* and delete what is found?

tonystone commented on Pod/Cache/Internal/Meta/WriteAheadLog.swift:
```diff
@@ -99,11 +99,11 @@ internal class WriteAheadLog {
         defer {
             objc_sync_exit(self)
         }
-        let sequenceNumberBlockStart = nextSequenceNumber;
```
You have something against semi-colons :-)

Commit:
Updating the logic to delete the persistent store files to not fail if one of the files does not exist.

tonystone commented on Pod/Cache/GenericCoreDataStack.swift:
```diff
@@ -211,6 +239,18 @@ public class GenericCoreDataStack<CoordinatorType: NSPersistentStoreCoordinator,
         }
     }
     
+    private func deleteIfExists(fileURL url: NSURL) {
```
deleteIfExists says delete it if it exists, and if it does not, do nothing, not even log an error.   The only file that is truly required to be present is the main file.  This is because we've already found the file and found that it's incompatible.

I believe I would make deleteIfExists check for the file first, then try the delete.  If successful, return, if the delete then throws an error for some reason, I would let it propagate to the caller.  I wouldn't log anything if the file does not exist.

tonystone commented on Pod/Cache/GenericCoreDataStack.swift:
```diff
@@ -185,14 +191,36 @@ public class GenericCoreDataStack<CoordinatorType: NSPersistentStoreCoordinator,
+                    logInfo(tag) { "Attempting to remove file \(storeURL) and -shm and -wal files" }
+                    if let path = storeURL.path {
+
+                        deleteIfExists(fileURL: storeURL)
```
See note below about deleteIfExists, seems this file should assume it present since it already found it to be incompatible.

Commit:
Inside `deleteIfExists(_)` making sure the file exists and then attempt to delete. Also re-throwing errors if NSFileManager fails.

Commit:
fixing conditional check

Commit:
Adding a requirements section to the README

Commit:
Creating a second test model for unit testing purposes.

Commit:
- using the static member on NSPersistentStoreCoordinator to fetch the metadata
- updating namingPrefix to storeNamePrefix and making it a mandatory parameter on GenericCoreDataStack

Commit:
updating Xcode touched files

tonystone commented:
Approved.
